dbms/src/Flash/Coprocessor/DAGExpressionAnalyzer.cpp

```diff
@@ -158,6 +158,34 @@ void DAGExpressionAnalyzer::fillArgumentDetail(
     arg_collators.push_back(removeNullable(arg_types.back())->isString() ? getCollatorFromExpr(arg) : nullptr);
 }
 
+void DAGExpressionAnalyzer::fillArgumentDetailForAggFuncBitwise(
+    const ExpressionActionsPtr & actions,
+    const tipb::Expr & arg,
+    Names & arg_names,
+    DataTypes & arg_types,
+    TiDB::TiDBCollators & arg_collators)
+{
+    DataTypePtr uint64_type = std::make_shared<DataTypeUInt64>();
+    const Block & sample_block = actions->getSampleBlock();
+    String name = getActions(arg, actions);
+    DataTypePtr orig_type = sample_block.getByName(name).type;
+    if (!removeNullable(orig_type)->equals(*uint64_type))
+    {
+        if (orig_type->isNullable())
+        {
+            name = appendCast(makeNullable(uint64_type), actions, name);
+        }
+        else
+        {
+            name = appendCast(uint64_type, actions, name);
+        }
+    }
+
+    arg_names.push_back(name);
+    arg_types.push_back(orig_type->isNullable() ? makeNullable(uint64_type) : uint64_type);
+    arg_collators.push_back(removeNullable(arg_types.back())->isString() ? getCollatorFromExpr(arg) : nullptr);
+}
+
```

Comment:
Can we separate the logic of appending cast and filling argument details?

Reply:
You mean making two independent methods? But I think the logic is strongly related with current `fillArgumentDetailForAggFuncBitwise`, and cannot be used by others.

Reply:
Sure! then maybe add some comments to indicate this function appends cast to expr actions and the corresponding reasons?

Reply:
Ok.